## G4-ChIP-seq analysis
Raw fastq reads from G4-ChIP-seq in K562 cells were trimmed with cutadapt50 to remove adapter sequences and low-quality reads (mapping quality < 10). Reads were aligned to the human genome (version hg19) with BWA51 and duplicates were removed using Picard. Peaks were called by MACS252 (p < 10-5) following our previous work31: https://github.com/sblab-bioinformatics/dna-secondary-struct-chrom-lands/blob/master/Methods.md 
Peaks were merged from different replicates with bedtools multiIntersect. Only peaks overlapping in 3 out 5 replicates were considered high-confidence. K562 datasets for DHSs (ENCSR000EPC), DNMT1 ChIP-seq (ENCSR987PBI), whole genome bisulfite sequencing (ENCSR765JPC) were downloaded from ENCODE. Promoters were defined as 1 kb (+/−) from the transcription start sites of 31,239 hg19 transcripts. Methylation levels at CpG sites with less than 5x coverage were discarded. If not otherwise specified, CGI34 were downloaded using the UCSC’s table browser and then ported to human genome release hg38 using the batch coordinate conversion (liftover) tool of the UCSC. The alternative CGI sets were generated using CpGCluster36. 
