
## DNA G-quadruplex structures mould the DNA methylome

Published computational analyses and scripts for the following manuscript:

*Include reference*

### Contents

- **BG4 ChIP-seq in K562 cells**, following our [previous work](https://github.com/sblab-bioinformatics/dna-secondary-struct-chrom-lands/blob/master/Methods.md)
  - Read processing, mapping and peak calling
  - CPG Islands
  - Methylation analysis
  - Enrichment analysis
  - Monte Carlo simulation

- [**Whole genome bisulfite sequencing in HaCaT cells**](wgbs_hacat.md)
  - [Read processing, alignment and de-duplication](wgbs_hacat.md#read-processing-alignment-and-de-duplication)
  - [Methylation extraction, aggregation and filter by coverage](wgbs_hacat.md#methylation-extraction-aggregation-and-filter-by-coverage)
  - [Analysis of methylation changes in BG4 ChIP-Seq peaks, CGIs and promoters before and after Entinostat treatment](wgbs_hacat.md#analysis-of-methylation-changes-in-bg4-chip-seq-peaks-cgis-and-promoters-before-and-after-entinostat-treatment)
  
